src/Playwright/Core/Route.cs

```diff
@@ -68,14 +72,34 @@ public async Task FulfillAsync(RouteFulfillOptions options = default)
                 options.Response).ConfigureAwait(false);
 
             await RaceWithPageCloseAsync(_channel.FulfillAsync(normalized)).ConfigureAwait(false);
+            ReportHandled(true);
         }
 
-        public Task AbortAsync(string errorCode = RequestAbortErrorCode.Failed) => RaceWithPageCloseAsync(_channel.AbortAsync(errorCode));
+        public async Task AbortAsync(string errorCode = RequestAbortErrorCode.Failed)
+        {
+            CheckNotHandled();
+            await RaceWithPageCloseAsync(_channel.AbortAsync(errorCode)).ConfigureAwait(false);
+            ReportHandled(true);
+        }
+
+        public async Task ContinueAsync(RouteContinueOptions options = default)
+        {
+            CheckNotHandled();
+            _request.ApplyFallbackOverrides(new RouteFallbackOptions().FromRouteContinueOptions(options));
+            await InnerContinueAsync().ConfigureAwait(false);
+            ReportHandled(true);
+        }
 
-        public Task ContinueAsync(RouteContinueOptions options = default)
+        internal async Task InnerContinueAsync(bool @internal = false)
         {
-            options ??= new RouteContinueOptions();
-            return RaceWithPageCloseAsync(_channel.ContinueAsync(url: options.Url, method: options.Method, postData: options.PostData, headers: options.Headers));
+            var options = _request.FallbackOverridesForContinue();
+            await _channel.Connection.WrapApiCallAsync(
+                async () =>
+                {
+                    await RaceWithPageCloseAsync(_channel.ContinueAsync(url: options.Url, method: options.Method, postData: options.PostData, headers: options.Headers)).ConfigureAwait(false);
+                    return 0;
```

Comment:
note: `return 0` is a hack so that the generic of WrapApiCallAsync knows what to return and does not yell, didn't find a better way.